modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/TWKBReader.java

```diff
@@ -220,18 +220,19 @@ private TWKBMetadata readMetadata() throws IOException {
         metadata.setHeader(header);
 
         int dims = 2;
+        // accordingto https://github.com/TWKB/Specification/blob/master/twkb.md it is Z then M
```

Comment:
```suggestion
        // according to https://github.com/TWKB/Specification/blob/master/twkb.md it is Z then M
```

So did you compare the JTS and GeoTools implementation? Does the GeoTools one have any improvement related to lower memory allocation or performance in general?

Uh, I tried to have a quick look, the difference seems to be pretty significant in general... in the loop that reads the coordianates I see the JTS code is computing a Math.pow for each and every ordinate value, that does not look so good performance-wise (the GT version computes those once and then the loops reuses them instead).
The JTS code however keeps track of previous values as longs, before the double expansion, which is likely more precise.

I'd say the merger of the two would require some work...